launch_ros/launch_ros/actions/node.py

```diff
@@ -172,6 +175,91 @@ def __init__(
+                value = param.get_attr('value', data_type=None, optional=True)
+                nested_params = param.get_attr('param', data_type=List[Entity], optional=True)
+                if value is not None and nested_params:
+                    raise RuntimeError('param and value attributes are mutually exclusive')
+                elif value is not None:
+                    def normalize_scalar_value(value):
+                        if isinstance(value, str):
+                            value = parser.parse_substitution(value)
+                            if len(value) == 1 and isinstance(value[0], TextSubstitution):
+                                value = value[0].text  # python `str` are not converted like yaml
+                        return value
+                    if isinstance(value, list):
+                        value = [normalize_scalar_value(x) for x in value]
+                    else:
+                        value = normalize_scalar_value(value)
+                    param_dict[name] = value
+                elif nested_params:
+                    param_dict.update({
+                        name: get_nested_dictionary_from_nested_key_value_pairs(nested_params)
+                    })
+                else:
+                    raise RuntimeError('either a value attribute or nested params are needed')
+            return param_dict
+
+        normalized_params = []
+        params_without_from = []
+        for param in params:
+            from_attr = param.get_attr('from', optional=True)
+            name = param.get_attr('name', optional=True)
+            if from_attr is not None and name is not None:
+                raise RuntimeError('name and from attributes are mutually exclusive')
+            elif from_attr is not None:
+                # 'from' attribute ignores 'name' attribute,
+                # it's not accepted to be nested,
+                # and it can not have children.
+                normalized_params.append(parser.parse_substitution(from_attr))
+                continue
+            elif name is not None:
+                params_without_from.append(param)
+                continue
+            raise ValueError('param Entity should have name or from attribute')
+        normalized_params.append(
+            get_nested_dictionary_from_nested_key_value_pairs(params_without_from))
+        return normalized_params
+
+    @classmethod
+    def parse(cls, entity: Entity, parser: Parser):
+        """Parse node."""
+        # See parse method of `ExecuteProcess`
+        _, kwargs = super().parse(entity, parser, 'args')
+        kwargs['arguments'] = kwargs['args']
+        del kwargs['args']
+        kwargs['node_name'] = kwargs['name']
+        del kwargs['name']
+        package = parser.parse_substitution(entity.get_attr('package'))
+        kwargs['package'] = package
+        executable = parser.parse_substitution(entity.get_attr('executable'))
+        kwargs['node_executable'] = executable
```

Comment:
@ivanpauno just curious, why the two step assignment?

Reply:
Just a bad habit.